components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/ConfigurationDAO.java

```diff
@@ -312,4 +312,16 @@ List getResourcesByType(int tenantId, String resourceTypeId)
      * @throws ConfigurationManagementException if an error occurs while validating the resourceId.
      */
     boolean isExistingResource(int tenantId, String resourceId) throws ConfigurationManagementException;
+
+    /**
+     * Delete all the resources by the given resource type in the given tenant.
+     *
+     * @param tenantId       Id of the tenant
+     * @param resourceTypeId Id of the {@link ResourceType}.
+     * @throws ConfigurationManagementException Configuration Management Exception.
+     */
+    default void deleteResourcesByType(int tenantId, String resourceTypeId) throws ConfigurationManagementException {
+
+        throw new ConfigurationManagementException("This method is not implemented", null);
```

Comment:
Why are we passing null here? IMO this is not a good practice

Reply:
Don't we need to throw a NotImplementedException here?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5030/commits/89229d99304bf28cb30b59a43d181d33054a0be6